php_phongo.c

```diff
@@ -1638,30 +1613,16 @@ static void php_phongo_cursor_iterator_rewind(zend_object_iterator *iter TSRMLS_
 {
 	php_phongo_cursor_iterator *cursor_it = (php_phongo_cursor_iterator *)iter;
 	php_phongo_cursor_t        *cursor = cursor_it->cursor;
+	const bson_t               *doc;
 
 	php_phongo_cursor_free_current(cursor);
 	cursor_it->current = 0;
```

Comment:
Since the cursor is initialized to point to its first result, the rewind handler really need not do anything other than load the current element. `cursor_it->current` is already initialized to zero when allocating the structure. 

I actually think we should not reset `cursor_it->current` to zero here, as rewinding should be a NOP. Alternatively, we could be more aggressive and throw an exception if `cursor_it->current` is greater than zero (to notify users that they shouldn't be rewinding after they've started iteration).

Whatever we decide to do, we can easily test by wrapping our Cursor in an IteratorIterator, which allows us to trigger the rewind handler by calling `rewind()` in userland.